hilti/src/compiler/parser/parser.yy

```diff
@@ -710,6 +711,7 @@ expr_e        : BEGIN_ '(' expr ')'              { $$ = hilti::expression::Unres
               | NEW expr                         { $$ = hilti::expression::UnresolvedOperator(hilti::operator_::Kind::New, {std::move($2), hilti::expression::Ctor(hilti::ctor::Tuple({}, __loc__))}, __loc__); }
               | NEW type                         { $$ = hilti::expression::UnresolvedOperator(hilti::operator_::Kind::New, {hilti::expression::Type_(std::move($2)), hilti::expression::Ctor(hilti::ctor::Tuple({}, __loc__))}, __loc__); }
               | NEW type '(' opt_exprs ')'       { $$ = hilti::expression::UnresolvedOperator(hilti::operator_::Kind::New, {hilti::expression::Type_(std::move($2)), hilti::expression::Ctor(hilti::ctor::Tuple(std::move($4), __loc__))}, __loc__); }
+              | TYPEINFO '(' type ')'            { $$ = hilti::expression::TypeInfo(std::move($3), __loc__); }
```

Comment:
WDYT about also allowing to pass an expression to `typeinfo`? Since not all types can be instantiated in scripts a type variant is needed, but I could imagine an expression variant being useful to for debugging or defensive programming.

Reply:
Good thought, adding `typeinfo(expr)`.